### ISSUE TITLE
Common weights/structure for language-specific pages

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/opentelemetry.io#2829]
As discussed during the SIG meeting 05/22/23 we want to have a common structure for all language navigations:

  - 10 Getting Started
  - 20 Automatic
    - 10 Configuration
    - 20 Extensions
    - ...
  - 30 Manual
  - 40 Libraries
  - 50 Exporters
  - 60 (Context) Propagation
  - 70 Resources
  - 80 Sampling
  - 100-200 *language specific docs*
  - 210 API (external link)
  - 220 Examples (external link)

I added the weights after the discussion to make it consistent through that

this also is going to fix #2431 

TODO:
- [x] #2831
- [x] #2830
- [x] #2832
- [x] Go:
  - [x] https://github.com/open-telemetry/opentelemetry-go/pull/4211
  - [x] #2849
- [x] #2833
- [x] #2834
- [x] #2835
- [x] #2836
- [x] #2837
- [x] Rust (not applicable)
- [x] #2838

**Comment by svrnm** [comment]
> My only suggestion would be to start language-specific pages at 100 instead of 90. Otherwise, great initiative, thanks!

good call, let me fix some of my PRs then ;-)

**Comment by svrnm** [comment]
@chalin , what do we need to do to get the latest updates for the go pages, since https://github.com/open-telemetry/opentelemetry-go/pull/4211 got merged?

**Comment by chalin** [comment]
@svrnm - refresh the Go submodule by running this command:

```sh
npm run update:submodule -- content-modules/opentelemetry-go
```

and submit the update as a PR.

**Comment by chalin** [comment]
Closed by #2849
